app/views/design-system/rsd-design-system/design-patterns/code-frame.html

```diff
@@ -0,0 +1,46 @@
+{% macro code(href, title, id, htmlCode, nunjucksCode) %}
+
+<div class="govuk-tabs govuk-!-margin-top-5 code-example" id="example-tabs-{{ id }}" data-module="govuk-tabs">
+  <h2 class="govuk-tabs__title">Contents</h2>
+  <ul class="govuk-tabs__list">
+    <li class="govuk-tabs__list-item govuk-tabs__list-item--selected">
+      <a class="govuk-tabs__tab" href="#example-tabs__panel-{{ id }}">Example</a>
+    </li>
+    <li class="govuk-tabs__list-item">
+      <a class="govuk-tabs__tab" href="#html-code-tabs__panel-{{ id }}">HTML</a>
+    </li>
+    <li class="govuk-tabs__list-item">
+      <a class="govuk-tabs__tab" href="#nunjucks-code-tabs__panel-{{ id }}">Nunjucks</a>
+    </li>
+  </ul>
+
+  <div class="govuk-tabs__panel" id="example-tabs__panel-{{ id }}">
+    <p>
+      <a href="{{ href }}" target="_blank">Open this example<span class="govuk-visually-hidden"> ({{title}})</span> in a new tab</a>
+    </p>
+    <iframe class="app-example__frame" data-module="app-iframe" src="{{ href }}" title="{{title}}" frameborder="0" loading="lazy"></iframe>
+  </div>
+
+  <div class="govuk-tabs__panel" id="html-code-tabs__panel-{{ id }}">
+    <pre data-module="app-copy" tabindex="0">
+      {{ htmlCode | safe }}
+    </pre>
+  </div>
+
+  {% if nunjucksCode %}
+  <div class="govuk-tabs__panel" id="nunjucks-code-tabs__panel-{{ id }}">
+    <pre data-module="app-copy" tabindex="0">
+      {{ nunjucksCode | safe }}
+    </pre>
+  </div>
+  {% else %}
+  <div class="govuk-tabs__panel" id="nunjucks-code-tabs__panel-{{ id }}">
+    We have not been able to create Nunjucks for this pattern.
+
+    If you have any problems implementing this pattern, ask in the <a href="https://ukgovernmentdfe.slack.com/archives/C04N7UJHRHU" target="_blank">#rsd-ucd-community-public</a> Slack channel. 
```

Comment:
all target="_blank" links need to say "Open in new tab" either hidden or visually and, must include the rel="noopener noreferrer" attribute

For links which only DfE staff can access (slack, intranet, devops) make sure it says it's for DfE staff only or similar.